components/org.wso2.carbon.identity.api.server.claim.management/org.wso2.carbon.identity.rest.api.server.claim.management.v1/src/main/java/org/wso2/carbon/identity/rest/api/server/claim/management/v1/core/ServerClaimManagementService.java

```diff
@@ -1810,6 +1835,95 @@ private void validateSubAttributeUpdate(LocalClaimReqDTO localClaimReqDTO) throw
+            return; // No input format specified, no validation needed.
+        }
+        InputTypeEnum inputTypeEnum = localClaimReqDTO.getInputFormat().getInputType();
+        DataTypeEnum dataType = localClaimReqDTO.getDataType();
+
+        switch (inputTypeEnum) {
+            case TEXT_INPUT:
+            case TEXTAREA:
+                break;
+            case DROPDOWN:
+            case RADIO_GROUP:
+            case MULTI_SELECT_DROPDOWN:
+            case CHECKBOX_GROUP: {
+                String inputTypeName = inputTypeEnum.getInputTypeName();
+                if (inputTypeEnum == InputTypeEnum.CHECKBOX_GROUP && DataTypeEnum.BOOLEAN.equals(dataType)) {
+                    break;
+                }
+                if (!DataTypeEnum.STRING.equals(dataType)) {
+                    handleInputFormatClientException("Input format: " + inputTypeName + " not aligned with string " +
+                            "data type.");
+                }
+                if (ArrayUtils.isEmpty(localClaimReqDTO.getCanonicalValues())) {
+                    handleInputFormatClientException("Input format: " + inputTypeName + " requires multiple options " +
+                            "to be defined.");
+                }
+                boolean isMultiValued = Boolean.TRUE.equals(localClaimReqDTO.getMultiValued());
+                if (inputTypeEnum == InputTypeEnum.DROPDOWN || inputTypeEnum == InputTypeEnum.RADIO_GROUP) {
+                    if (isMultiValued) {
+                        handleInputFormatClientException("Input format: " + inputTypeName + " requires multi valued " +
+                                "property to be false.");
+                    }
+                } else {
+                    if (!isMultiValued) {
+                        handleInputFormatClientException("Input format: " + inputTypeName + " requires multi valued " +
+                                "property to be enabled.");
+                    }
+                }
+                break;
+            }
+            case DATE_PICKER:
+                if (!DataTypeEnum.DATE_TIME.equals(dataType)) {
+                    handleInputFormatClientException("Input format: date_picker can be used with date data type.");
+                }
+                break;
+            case NUMBER_INPUT:
+                if (!DataTypeEnum.INTEGER.equals(dataType)) {
+                    handleInputFormatClientException("Input format: number_input can be used with integer data type.");
+                }
+                break;
+            case TOGGLE:
+                if (!DataTypeEnum.BOOLEAN.equals(dataType)) {
+                    handleInputFormatClientException("Input format: toggle should be boolean data type.");
+                }
+                break;
+        }
+    }
+
+    private void handleInputFormatClientException(String errorDescription) throws ClaimMetadataClientException {
```

Comment:
Without defining different error codes, same error code is returned with different error descriptions based on the validation failures belong to the same category (e.g., input format validation).
Need to identify the proper way to handle.